File: jvm/src/main/java/edu/kit/kastel/vads/compiler/ir/util/DebugInfo.java
package edu.kit.kastel.vads.compiler.ir.util;

import de.mr_pine.c0ne.Span;

/// Provides information to ease debugging
public sealed interface DebugInfo {
    enum NoInfo implements DebugInfo {
        INSTANCE
    }

    record SourceInfo(Span span) implements DebugInfo {}
}

File: jvm/src/main/java/de/mr_pine/c0ne/ir/util/DebugInfo.kt
package de.mr_pine.c0ne.ir.util

import de.mr_pine.c0ne.Span

/** Provides information to ease debugging */
interface DebugInfo {
    enum class NoInfo : DebugInfo {
        INSTANCE
    }

    data class SourceInfo(val span: Span) : DebugInfo
}
